lexerState.go

```diff
@@ -57,6 +58,7 @@ var validLexerStates = []lexerState{
 			COMPARATOR,
 			LOGICALOP,
 			CLAUSE_CLOSE,
+			TERNARY,
```

Comment:
Oh, true. I'm used to putting parens around things like that, so I didn't think of that. But you're correct, that's a totally valid case. No problem then!